target/linux/ramips/image/common-tp-link.mk

```diff
@@ -53,6 +53,20 @@ define Device/tplink-v1-okli
 	append-metadata
 endef
 
+define Device/tplink-v2-lzma-custom
+  DEVICE_VENDOR := TP-Link
+  TPLINK_FLASHLAYOUT :=
+  TPLINK_HWID :=
+  TPLINK_HWREV := 0x1
+  TPLINK_HWREVADD := 0x0
+  TPLINK_HVERSION := 3
+  KERNEL := kernel-bin | append-dtb | lzma -d22
+  KERNEL_INITRAMFS := kernel-bin | append-dtb
+  IMAGES += sysupgrade.bin
```

Comment:
Why do you add `sysupgrade.bin`, it should be already in this list. 
Is the `factory.bin` image created and working? I think you have to add it here too or remove the `IMAGE/sysupgrade.bin` line.